cli/cmd/sync_v3.go

```diff
@@ -345,53 +345,66 @@ func syncConnectionV3(ctx context.Context, source v3source, destinations []v3des
 			return fmt.Errorf("unknown message type: %T", m)
 		}
 	}
+
 	err = syncClient.CloseSend()
 	if err != nil {
 		return err
 	}
 	totals := sourceClient.Metrics()
-
-	for i := range destinationsClients {
-		if destinationSpecs[i].WriteMode == specs.WriteModeOverwriteDeleteStale {
-			if err := deleteStale(writeClients[i], tablesForDeleteStale, sourceName, syncTime); err != nil {
-				return err
-			}
-		}
-		if _, err := writeClients[i].CloseAndRecv(); err != nil {
-			return err
-		}
-		if _, err := destinationsPbClients[i].Close(ctx, &plugin.Close_Request{}); err != nil {
-			return err
-		}
-	}
-
 	sourceWarnings := totals.Warnings
 	sourceErrors := totals.Errors
 	syncSummaries := make([]syncSummary, len(destinationsClients))
 	for i := range destinationsClients {
 		m := destinationsClients[i].Metrics()
-		totals.Warnings += m.Warnings
-		totals.Errors += m.Errors
 		syncSummaries[i] = syncSummary{
 			Resources:           uint64(totalResources),
 			SourceErrors:        sourceErrors,
 			SourceWarnings:      sourceWarnings,
 			SyncID:              uid,
+			SyncTime:            syncTime,
 			SourceName:          sourceSpec.Name,
 			SourceVersion:       sourceSpec.Version,
 			SourcePath:          sourceSpec.Path,
-			CliVersion:          Version,
+			CLIVersion:          Version,
 			DestinationErrors:   m.Errors,
 			DestinationWarnings: m.Warnings,
 			DestinationName:     destinationSpecs[i].Name,
 			DestinationVersion:  destinationSpecs[i].Version,
 			DestinationPath:     destinationSpecs[i].Path,
 		}
 	}
-	err = persistSummary(summaryLocation, syncSummaries)
-	if err != nil {
+	if err := persistSummary(summaryLocation, syncSummaries); err != nil {
 		log.Warn().Err(err).Msg("Failed to persist sync summary")
 	}
+	for _, summary := range syncSummaries {
+		log.Info().Interface("summary", summary).Msg("Sync summary")
+		for i := range destinationsClients {
+			if !destinationSpecs[i].SyncSummary {
+				continue
+			}
+			// Only send the summary to the destination that matches the current destination
+			if destinationSpecs[i].Name != summary.DestinationName || destinationSpecs[i].Version != summary.DestinationVersion || destinationSpecs[i].Path != summary.DestinationPath {
+				continue
+			}
+			if err := sendSummary(writeClients[i], destinationSpecs[i], destinationsClients[i], destinationTransformers[i], &summary, noMigrate); err != nil {
```

Comment:
Let's say we have an error sending the summary to the first destination? Shouldn't we try sending the summary for the next ones?

Maybe we can collect all errors via `errors.Join` then report the ones that failed